### DISCUSSION TITLE
How can you cross-reference a table written in raw LaTeX?

### DISCUSSION THREAD

**Discussion by maxdrohde** [quarto-dev/quarto-cli#6734] [Q&A] [answered]
### Description

I'm trying to reference a table using the Quarto `@` syntax, where I have written the table code in raw LaTeX. Below are two approaches I tried, and neither seem to work. Any help is appreciated!

 - Using the `\label{}` commend in LaTeX
 - Adding `#| label:` to the cell

````

---
title: "Table references"
format: pdf
---

# First try

```{=latex}
\begin{table}[h!]
\centering
\begin{tabular}{||c c||} 
 \hline
 Col1 & Col2 \\
 \hline\hline
 1 & 4 \\ 
 2 & 5 \\
 3 & 6 \\
 \hline
\end{tabular}
\caption{Test caption}
\label{tbl-example}
\end{table}
```

This table is called @tbl-example.

# Second try

```{=latex}
#| label: tbl-example

\begin{table}[h!]
\centering
\begin{tabular}{||c c||} 
 \hline
 Col1 & Col2 \\
 \hline\hline
 1 & 4 \\ 
 2 & 5 \\
 3 & 6 \\
 \hline
\end{tabular}
\caption{Test caption}
\end{table}
```

This table is called @tbl-example.
````

**Comment by cscheid** [comment]
If you use a raw latex entry with `\label{}`, then you need to use a raw latex reference as well, with `` `\ref{tbl-example}`{=latex} ``. (It might even be the case that Pandoc recognizes `\ref{tbl-example}` directly.)

**Reply by maxdrohde** [reply]
@cscheid Thanks! I'd prefer not to have to use `label{}` if there's a way to do it with Quarto's syntax -- is this the only option when the table is in LaTeX?

**Reply by cscheid** [reply, marked as answer]
https://github.com/quarto-dev/quarto-cli/pull/6620 will change things. It both has some support for attempting to pull labels from raw latex captions (not very robust, because it's not something that can be solved in full generality) and you'll be able to call anything you want a table, so something like

````

::: {#tbl-1}

```{=latex}
% Or whatever actual you want, includegraphics, whatever. 
% (_not_ \begin{table}, just the content of it)
\begin{tabular}{lll}
\end{tabular}
```

This is a table caption.

:::

See @tbl-1
````

will work.

**Reply by baris-ari** [reply]
This almost worked for me, but it created an unexpected change by putting a second caption "(a)". 

```

---
title: "Table reference"
format: pdf
---

::: {#tbl-1}
```{=latex}
\begin{tabular}{||c c||} 
 \hline
 Col1 & Col2 \\
 \hline\hline
 1 & 4 \\ 
 2 & 5 \\
 3 & 6 \\
 \hline
\end{tabular}

This is a table caption.
:::

See @tbl-1

```

![Screenshot 2024-01-12 at 22 56 49](https://github.com/quarto-dev/quarto-cli/assets/127899053/25227dcc-ac4e-4f09-ba10-1e9cfff00dfc)

I am currently using \ref{tbl-example} as suggested, which is recognized by Pandoc.  I am confused because this second caption does not appear when I don't use the `::: {#tbl-1}` route.

**Reply by mcanouil** [reply]
@baris-ari which Quarto version are you using?

**Reply by baris-ari** [reply]
Version: 1.3.450

**Reply by baris-ari** [reply]
I updated and problem gone. Thank you!

**Reply by koichiosamura** [reply]
@cscheid Sorry for bringing back this issue. I had a similar problem with the table (the pdf output brought that "(a)" subcaption) but it disappeared once I updated Quarto.

But now I am having a similar issue with a Mermaid diagram. I am pasting the code below, alongside the image.


```

---
title: "Figure test"
format: pdf
---

::: {#fig-test}


{mermaid}
%%| fig-height: 0.8
%%| label: fig-test 

  graph LR
    A(A) -->|move right| B(B)
    B -->|move left| A
    style A fill:white,stroke:black,stroke-width:2px
    style B fill:white,stroke:black,stroke-width:2px

This is a test diagram.
:::

```




![Screenshot 2024-04-06 at 22 29 38](https://github.com/quarto-dev/quarto-cli/assets/101427341/4c34ee6c-774e-48b5-9d9d-ac11a07a14ce)

I checked other forums and found that at least in a regular R chunk the option `fig-subcap: false` should work, but this does not happen with Mermaid.

Any suggestions? Thanks a lot.

**Reply by mcanouil** [reply]
Could you actually make your code valid? Currently, it's not (or ever) going to work with the current code shared.

Note also that you are setting twice the exact same id/label.
Label/id should be unique.
You can't set the id of the fenced div using the id you gave to the mermaid code.

---

You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

If you have multiple files (and if it is **absolutely required** to have multiple files), please share as a Git repository.

<table>
<tr><th>R</th><th>Python</th></tr>
<tr><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document.

```{r}
x <- c(1, 2, 3, 4, 5)
y <- c(1, 4, 9, 16, 25)

plot(x, y)
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

This is a reproducible Quarto document.

```{python}
import matplotlib.pyplot as plt

x = [1, 2, 3, 4, 5]
y = [1, 4, 9, 16, 25]

plt.plot(x, y)
plt.show()
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td></tr>
</table>

Additionally and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.

**Reply by koichiosamura** [reply]
Thank you, it worked. I think the issue was the double label. 

As for the backticks, I removed it because some how it was breaking my reply to this thread. I was not aware about using four instead of three. But thank you.
